langchain/chains/sql_database/base.py

```diff
@@ -50,6 +50,10 @@ class SQLDatabaseChain(Chain):
     to fix the initial SQL from the LLM."""
     query_checker_prompt: Optional[BasePromptTemplate] = None
     """The prompt template that should be used by the query checker"""
+    sql_cmd_parser: Optional[BaseOutputParser] = None
```

Comment:
hm we can already add an output parser to the llm chain, would that work instead?

Reply:
Extracting embedding for words takes too much vectors and query checker will not be able to check that long sql. Putting the parser here will make the chain have direct control of the parser.

Also, for better readability we think dumping direct result from LLM is better than the parsed one.

For example 
```sql
SELECT * FROM table ORDER BY DISTANCE(vector, NeuralArray(langchain)) LIMIT 5
```
Or you have to dealing with
```sql
SELECT * FROM table ORDER BY distance(vector, [0.2314, 0.132415, ..., 0.13941] LIMIT 5
```
which does not make any sense to human.

Reply:
There is another implementation can inject parsers into the llm_chain:
1. define a `SQLCommand` object to store both the parsed and unparsed llm output
2. force the llm_chains to always produce a SQLCommand object for further use of the raw output from LLM

This is a probable approach as far as I can think of. Which one do you prefer?

Reply:
Revised with a dictionary that preserved output from LLM.